src/XMakeTasks/Microsoft.CSharp.targets

```diff
@@ -17,6 +17,15 @@ Copyright (C) Microsoft Corporation. All rights reserved.
 
 <Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
    <!-- 
+        We are doing a cross-targeting traversal build if there is no list of target frameworks specified
+        nor is there a current target framework being built individually. In that case, this import is
+        redirected to Microsoft.Common.Traversal.targets.
+   -->
+   <PropertyGroup Condition="'$(TargetFrameworks)' != '' and '$(TargetFramework)' == ''">
```

Comment:
IF the single target framework is specified in TargetFrameworks, then yes it loops. You can also configure yourself as an inner build and we won't loop, but then there should be no TargetFrameworks with an s and this condition works too.